src/ripple/basics/Slice.h

```diff
@@ -120,6 +125,27 @@ Stream& operator<<(Stream& s, Slice const& v)
     return s;
 }
 
+template <class T, class Alloc>
+std::enable_if_t<
+    std::is_same<T, char>::value ||
+        std::is_same<T, unsigned char>::value,
```

Comment:
I don't see a need for that. Quite the opposite, I'd prefer that the vocabulary of types we use as containers for contiguous memory be small and well defined, instead of open ended and numerous.